src/components/FlyoutMenu/FlyoutMenu.jsx

```diff
@@ -0,0 +1,223 @@
+              className={`${iotPrefix}-flyout-menu__cancel`}
+              kind="secondary"
+              onClick={() => {
+                if (open === undefined) {
+                  setIsOpen(false);
+                }
+
+                onCancel();
+              }}
+              aria-label={i18n.cancelButton}
+            >
+              {i18n.cancelButton}
+            </Button>
+            <Button
+              className={`${iotPrefix}-flyout-menu__submit`}
+              aria-label={i18n.applyButton}
+              onClick={() => {
+                if (open === undefined) {
+                  setIsOpen(false);
+                }
+
+                onApply();
+              }}
+            >
+              {i18n.applyButton}
+            </Button>
+          </div>
+        ) : null}
+      </Tooltip>
+    );
+  };
+
+  return (
+    <div>
+      {renderButton ? (
+        <Button
+          aria-label={iconDescription}
+          iconDescription={iconDescription}
+          className={`${buttonProps.className ??
+            buttonProps.className} ${iotPrefix}-flyout-menu__trigger-button ${
+            light ? `${iotPrefix}-flyout-menu--light` : ''
+          }${open || isOpen ? ` ${iotPrefix}-flyout-menu--open` : ''}`}
+          disabled={disabled}
+          kind="ghost"
+          {...buttonProps}
+          onClick={() => {
+            setIsOpen(!isOpen);
+          }}
+        >
+          {renderTooltip()}
+        </Button>
+      ) : null}
+
+      {renderButton ? null : renderTooltip()}
+    </div>
+  );
+};
+
+const propTypes = {
+  tooltipId: PropTypes.string,
+
+  /**
+   * The ID of the trigger button.
+   */
+  triggerId: PropTypes.string,
+
+  /**
+   * Optional starting value for uncontrolled state
+   */
+  defaultOpen: PropTypes.bool,
+
+  /**
+   * Open/closed state.
+   */
+  open: PropTypes.bool,
+
+  /**
+   * Contents to put into the flyout menu.
+   */
+  children: PropTypes.node,
+
+  /**
+   * The CSS class names of the flyout menu.
+   */
+  className: PropTypes.string,
+
+  /**
+   * The CSS class names of the trigger UI.
+   */
+  triggerClassName: PropTypes.string,
+
+  /**
+   * Where to put the flyout menu, relative to the trigger UI.
+   */
+  direction: PropTypes.oneOf(['bottom', 'top', 'left', 'right']),
+
+  /**
+   * The description of the default flyout icon
+   */
+  iconDescription: PropTypes.string.isRequired,
+
+  /**
+   * Optional prop to specify the tabIndex of the Tooltip
+   */
+  tabIndex: PropTypes.number,
+
+  /**
+   * Optional prop to specify the test id of the flyout
+   */
+  testId: PropTypes.string,
+
+  /**
+   * Renders cancel and apply button group
+   */
+  transactional: PropTypes.bool,
+
+  /**
+   * On Cancel button callback
+   */
+  onCancel: PropTypes.func.isRequired,
+
+  /**
+   * On Apply button callback
+   */
+  onApply: PropTypes.func.isRequired,
+
+  i18n: PropTypes.shape({
+    cancelButton: PropTypes.string,
+    applyButton: PropTypes.string,
```

Comment:
We can fully qualify these as `text`. In other places we qualify input `label`s as such, aria-labels as `aria` etc.
```suggestion
    cancelButtonText: PropTypes.string,
    applyButtonText: PropTypes.string,
```